src/v/coproc/event_listener.cc

```diff
@@ -252,4 +159,13 @@ event_listener::poll_topic(model::record_batch_reader::data_t& events) {
                                  : ss::stop_iteration::no;
 };
 
+ss::future<> event_listener::process_events(
+  absl::btree_map<script_id, parsed_event> events, model::offset last_offset) {
+    try {
+        co_await nodejs_handler.process(std::move(events));
+    } catch (event_handler_exception& ex) {
+        _offset = last_offset;
```

Comment:
if process throws because the script pushed was bad, then resetting to offset 0 won't solve anything, this will be an infinite loop of failures.

For me, setting offset to 0 is part of the failure recovery process for coproc, when we are not sure if the same coprocessors that exist in JS sidecar match with redpanda pacemaker, we wipe all state, and try rehydrate from the first offset

Reply:
Yes, I added switch in process events

Reply:
done